### ISSUE TITLE
CLI should allow user to specify web action extension when creating api

### ISSUE THREAD

**Issue by csantanapr** [openwhisk/openwhisk#2166]
the CLI when creating one route it should allow the user to specify the action name with no extension (which is the same at .http) or include the extension at the end of the action name for example:

wsk api create /basepath /path verb FQNAction[.extension]

Translate to following examples:
```
wsk api create /store /pets get myaction
```
With no extension this is the same as .http

wsk api create /store /pets get myaction.http
This is being explicit to use .http
```
wsk api create /store /pets post myaction.json
```
```
wsk api create /store /pets get myaction.html
```

And son on for the other extensions 

The backend today doesn't provide a way to know the list of valid extensions, this would be nice to have for the CLI or CLI helper actions to use to validate that the user used one of the valid actions.
For now I think is OK to hardcode the list of valid extensions in the CLI or CLI helper actions.

**Comment by csantanapr** [comment]
After discussing with @mdeuser over the design on how to specify here the final proposal

1. have an optional flag to specify the content type extension for the response when forming the web action url to be use by the API Gateway
2. change the default to .json extension
```
  --response-type CONTENT_TYPE_EXTENSION possible values json, http, text, svg (default json)
```

```
$ wsk api create -h

Usage:
  wsk api create ([BASE_PATH] API_PATH API_VERB ACTION] [--response-type EXTENSION] | --config-file CFG_FILE)  [flags]

Flags:
  -n, --apiname string         Friendly name of the API; ignored when CFG_FILE is specified (default BASE_PATH)
  -c, --config-file CFG_FILE   CFG_FILE containing API configuration in swagger JSON format
  --response-type EXTENSION EXTENSION with possible values json, http, text, svg (default .json)

```
This will allow the following examples:
```
wsk api create /store /pets get myaction
 (This default to --response-type json)
wsk api create /store /pets get myaction --response-type json
wsk api create /store /pets post myaction --response-type http
wsk api create /store /pets get myaction --response-type html
wsk api create /store /pets get myaction --response-type svg
```

**Comment by akrabat** [comment]
Presumably this means that if an `http` extension is used, the API Gateway would pass on the headers and status code provided by the action?

**Comment by csantanapr** [comment]
That's correct

**Comment by csantanapr** [comment]
Closed by #2149
